openmdao/core/driver.py

```diff
@@ -804,3 +854,113 @@ def _setup_simul_coloring(self, mode='fwd'):
                     del dvdict[dv]
                     dv = prom2abs[dv][0]
                     dvdict[dv] = col_dict
+
+    def _pre_run_model_debug_print(self):
+        """
+        Optionally print some debugging information before the model runs.
+        """
+        if not self.options['debug_print']:
+            return
+
+        if not MPI or MPI.COMM_WORLD.rank == 0:
+            header = 'Driver debug print for iter coord: {}'.format(
+                get_formatted_iteration_coordinate())
+            print(header)
+            print(len(header) * '-')
+
+        if 'desvars' in self.options['debug_print']:
+            desvar_vals = self.get_design_var_values()
+            if not MPI or MPI.COMM_WORLD.rank == 0:
+                print("Design Vars")
+                if desvar_vals:
+                    for name, value in iteritems(desvar_vals):
+                        print("{}: {}".format(name, value))
+                else:
+                    print("None")
+                print()
+
+    def _post_run_model_debug_print(self):
+        """
+        Optionally print some debugging information after the model runs.
+        """
+        if 'nl_cons' in self.options['debug_print']:
+            cons = self.get_constraint_values(lintype='nonlinear')
+            if not MPI or MPI.COMM_WORLD.rank == 0:
+                print("Nonlinear constraints")
+                if cons:
+                    for name, value in iteritems(cons):
+                        print("{}: {}".format(name, value))
+                else:
+                    print("None")
+                print()
+
+        if 'ln_cons' in self.options['debug_print']:
+            cons = self.get_constraint_values(lintype='linear')
+            if not MPI or MPI.COMM_WORLD.rank == 0:
+                print("Linear constraints")
+                if cons:
+                    for name, value in iteritems(cons):
+                        print("{}: {}".format(name, value))
+                else:
+                    print("None")
+                print()
+
+        if 'objs' in self.options['debug_print']:
+            objs = self.get_objective_values()
+            if not MPI or MPI.COMM_WORLD.rank == 0:
+                print("Objectives")
+                if objs:
+                    for name, value in iteritems(objs):
+                        print("{}: {}".format(name, value))
+                else:
+                    print("None")
+                print()
+
+
+class RecordingDebugging(Recording):
+    """
+    A class that acts as a context manager.
+
+    Handles doing the case recording and also the Driver
+    debugging printing.
+    """
+
+    def __init__(self, name, iter_count, recording_requester):
```

Comment:
\_\_init\_\_ does nothing except call the base class, so why is it defined here?

Reply:
fixed